### ISSUE TITLE
TypeError: this.setPickable is not a function

### ISSUE THREAD

**Issue by phet-steele** [phetsims/graphing-lines#83]
From phetsims/scenery/issues/629. Looks like the playStateObserver gets a notification queued somehow before it gets unlinked.

```
TypeError: this.setPickable is not a function
    at GraphPointSlopeNode.set pickable [as pickable] (https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/scenery/js/nodes/Node.js?bust=1499444125007:2876:34)
    at Array.playStateObserver (https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/graphing-lines/js/linegame/view/GraphTheLineNode.js?bust=1499444125007:135:31)
    at Emitter.emit2 (https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/axon/js/Emitter.js?bust=1499444125007:147:49)
    at Property._notifyObservers (https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/axon/js/Property.js?bust=1499444125007:207:29)
    at Property._setAndNotifyObservers (https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/axon/js/Property.js?bust=1499444125007:196:14)
    at Property.set (https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/axon/js/Property.js?bust=1499444125007:140:16)
    at GamePhaseProperty.hook (https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/graphing-lines/js/linegame/model/BaseGameModel.js?bust=1499444125007:82:34)
    at GamePhaseProperty.set (https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/graphing-lines/js/linegame/model/BaseGameModel.js?bust=1499444125007:291:12)
    at https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/graphing-lines/js/linegame/view/SettingsNode.js?bust=1499444125007:143:33
    at https://bayes.colorado.edu/continuous-testing/snapshot-1499443514075/sun/js/buttons/PushButtonModel.js?bust=1499444125007:168:9
Approximately 7/7/2017, 10:05:14 AM
```

**Comment by jonathanolson** [comment]
Looks like a "use after disposal" for Node.

**Comment by pixelzoom** [comment]
Assertion failure is in GraphTheLineNode, line 135.    I suspect that a similar failure would occur in MakeTheEquationNode, line 126.

**Comment by pixelzoom** [comment]
This is a little more complicated than I thought.  Moving unlink before dispose (as proposed above) doesn't fix the problem because `playStateObserver` is already queued to be called before `challengeNode.dispose` is called.

**Comment by pixelzoom** [comment]
Fixed in the above commit.  

@phet-steele please verify in master using `?ea&screens=4&fuzzMouse`.

FYI, I checked the current release branch (1.2) and this problem does not exist there.  So no need to publish a maintenance release, this is purely a fix in master.

**Comment by phet-steele** [comment]
Verified, bayes also has shown no errors since the fix.

**Comment by pixelzoom** [comment]
Reopening. 

This workaround can be removed after phetsims/scenery#649 is done.

**Comment by pixelzoom** [comment]
@phet-steele This has been fixed in phetsims/scenery#649, and the workaround in graphing-lines has been reverted.  Tested with `?ea&fuzzMouse&screens=4`. Please pull scenery and graphing-lines to re-verify.

**Comment by phet-steele** [comment]
Fuzzed master for an hour and saw no errors, closing.
